src/extra/others/snapshot/lv_snapshot.c

```diff
@@ -138,6 +143,12 @@ lv_res_t lv_snapshot_take_to_buf(lv_obj_t * obj, lv_img_cf_t cf, lv_img_dsc_t *
     obj->parent = screen;
 
     disp->inv_p = 0;
+
+    obj->coords.x2 = lv_area_get_width(&obj->coords) - 1;
+    obj->coords.x1 = 0;
```

Comment:
It should start on `ext_size` else the shadow will be clipped.

Where do you restore the original coordinates?

Reply:
I have corrected coords start value to ext_size.

Originally, I just mark obj layout as dirty and invalidate it. I have updated code to backup original coords and restore it at end.